### ISSUE TITLE
False positive for unused alias when doing interop

### ISSUE THREAD

**Issue by borkdude** [clj-kondo/clj-kondo#1840]
<img width="551" alt="Screenshot 2022-10-18 at 15 33 33" src="https://user-images.githubusercontent.com/284934/196444469-b877f86f-490b-45c5-a7fb-8bbe364c6f36.png">

cc @NoahTheDuke

**Comment by NoahTheDuke** [comment]
Thanks for the tag! Is there a meaningful difference between `sci.lang.Var.` and `lang.Var.` when `lang` is an alias?

**Comment by borkdude** [comment]
Yes. Just pinging you since you might have an interest in fixing it?

``` clojure
 $ clj
Clojure 1.11.0
user=> (ns foo) (defrecord FooVar [])
nil
foo.FooVar
foo=> (ns bar)
nil
bar=> (ns bar (:require [foo :as f]))
nil
bar=> (f.FooVar.)
Syntax error (ClassNotFoundException) compiling new at (REPL:1:1).
f.FooVar
bar=> (foo.FooVar.)
#foo.FooVar{}
```

**Comment by NoahTheDuke** [comment]
Hmm, is it possible to determine if a call is interop just from the token? From my in-progress tests:

```
<token: i/Foo>
<token: interop-helper/Foo>
```

**Comment by borkdude** [comment]
If there is no slash, then it's interop

**Comment by NoahTheDuke** [comment]
In that case, I can't get it to error because there's no namespace in the symbol and thus it doesn't go into the "lint-aliased-namespaces" branch. Do you have a minimal repro?

**Comment by borkdude** [comment]
Not minimal but the above screenshot (yeah I know) comes from the babashka/sci project.

**Comment by NoahTheDuke** [comment]
If I run `clj-kondo --lint src/sci/impl/utils.cljc --config "{:linters {:aliased-namespace-symbol {:level :warning}}}"` then I get the warning, so I'm pretty confused why it's not working in my test file.

**Comment by NoahTheDuke** [comment]
That works, but why/how/wtf lmao. Why would cljc change the symbols?

**Comment by borkdude** [comment]
Because when you write `sci.lang.Var.` in CLJS, it is in fact a usage of the `sci.lang` namespace and clj-kondo probably has some logic for this in `clj-kondo.impl.namespace/resolve` to mark it as such , but when you change it to `sci.lang/Var` it becomes invalid in CLJ. So this is definitely an edge case. We should try to suppress the warning in CLJS here, when inside a  CLJC file.

**Comment by NoahTheDuke** [comment]
Cool, thanks for all the help.
